compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/Types.java

```diff
@@ -172,6 +174,7 @@ public static Types getInstance(CompilerContext context) {
     public Types(CompilerContext context) {
         context.put(TYPES_KEY, this);
 
+        this.semanticAnalyzer = SemanticAnalyzer.getInstance(context);
```

Comment:
Types is like a util class. We shouldn't have a semantic analyzer instance here.

Reply:
We can move `constructNumericLiteralFromUnaryExpr` here.